src/AccelStepperFirmata.cpp

```diff
@@ -0,0 +1,431 @@
+            want acceleration we just set the acceleration value high enough so
+            that the chosen speed will be realized by the first step of a movement.
+            This simplifies some of the logic in StepperFirmata and gives us more flexibility
+            should an alternative stepper library become available at a future date.
+          */
+          if (decodedAcceleration == 0.0) {
+            stepper[deviceNum]->setAcceleration(MAX_ACCELERATION);
+          } else {
+            stepper[deviceNum]->setAcceleration(decodedAcceleration);
+          }
+          /*
+            </HACK>
+          */
+        }
+
+      }
+
+      else if (stepCommand == STEPPER_SET_SPEED) {
+        // Sets the maxSpeed for accelStepper. We do not use setSpeed here because 
+        // all instances of accelStepper that have been created by firmata are
+        // using acceleration. More info about this hack in STEPPER_SET_ACCELERATION.
+        float speed = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setMaxSpeed(speed);
+        }
+
+      }
+
+      else if (stepCommand == MULTISTEPPER_CONFIG) {
+        
+        if (!group[deviceNum]) {
+          numGroups++;
+          group[deviceNum] = new MultiStepper();
+        }
+
+        for (byte i = index; i < argc; i++) {
+          byte stepperNumber = argv[i];
+          
+          if (stepper[stepperNumber]) {
+            groupStepperCount[deviceNum]++;
+            group[deviceNum]->addStepper(*stepper[stepperNumber]);
+          }
+        }
+
+        groupIsRunning[deviceNum] = false;
+      }
+
+      else if (stepCommand == MULTISTEPPER_TO) {
+
+        groupIsRunning[deviceNum] = true;
+        long positions[groupStepperCount[deviceNum]];
+        
+        for (byte i = 0, offset = 0; i < groupStepperCount[deviceNum]; i++) {
+          offset = index + (i * 5);
+          positions[i] = decode32BitSignedInteger(argv[offset], argv[offset+1], argv[offset+2], argv[offset+3], argv[offset+4]);
+        }
+
+        group[deviceNum]->moveTo(positions);
+        
+      }
+
+      else if (stepCommand == MULTISTEPPER_STOP) {
+
+        groupIsRunning[deviceNum] = false;
+        
+      }
+    }
+    return true;
+    
+  }
+  return false;
+}
+
+/*==============================================================================
+ * SETUP()
+ *============================================================================*/
+
+void AccelStepperFirmata::reset()
+{
+  for (byte i = 0; i < MAX_STEPPERS; i++) {
+    if (stepper[i]) {
+      free(stepper[i]);
+      stepper[i] = 0;
+    }
+  }
+  numSteppers = 0;
+  
+  for (byte i = 0; i < MAX_GROUPS; i++) {
+    if (group[i]) {
+      free(group[i]);
+      group[i] = 0;
+    }
+  }
+  numGroups = 0;
+}
+
+/*==============================================================================
+ * Helpers
```

Comment:
> Also, is endianness only an issue with longs or could that also be an issue with ints for example?

It affects both, however I would be hesitant to "fix" this until it presents itself. We assume the Arduino is big endian at the byte level and in the protocol we send bytes little endian.

If you look at the [protocol](https://github.com/firmata/protocol/blob/master/protocol.md#extended-analog), it shows the bytes going out little endian. However, if you look to the implementation you can see we assume bytes are ordered "76543210", because send bytes as (byte & 0x7F) and ((byte >> 7) & 0x01).